tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/VotingCompletion.java

```diff
@@ -0,0 +1,223 @@
+package org.unicode.cldr.web.api;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.HeaderParam;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+
+import org.eclipse.microprofile.openapi.annotations.Operation;
+import org.eclipse.microprofile.openapi.annotations.media.Content;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponse;
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponses;
+import org.eclipse.microprofile.openapi.annotations.tags.Tag;
+import org.unicode.cldr.util.CLDRFile;
+import org.unicode.cldr.util.CLDRLocale;
+import org.unicode.cldr.util.Level;
+import org.unicode.cldr.util.PathHeader;
+import org.unicode.cldr.util.PathHeader.Factory;
+import org.unicode.cldr.util.SupplementalDataInfo;
+import org.unicode.cldr.web.CookieSession;
+import org.unicode.cldr.web.DBUtils;
+import org.unicode.cldr.web.SurveyMain;
+import org.unicode.cldr.web.api.VotingParticipation.VotingParticipationResults;
+
+@Path("/completion")
+@Tag(name = "completion", description = "APIs for voting completion statistics")
+public class VotingCompletion {
```

Comment:
This is the new back end API. It can be tested at openapi/ui/ though it isn't used by the Survey Tool front end yet.